### PR TITLE
add support for importing/exporting NAPTR records

### PR DESCRIPTION
**Pull request by wfaulk**
Please when making a pull request:

- make sure changes are up to date with master.
- please use brief, descriptive commit messages.
- check code has been go formatted with 'go fmt'.
- ensure you've added an integration test (under internal/features) or a unit test.
- check the tests pass (make test).

cli53 has very good existing test coverage, so I'm unlikely to accept a pull request without
a test, but please do ping me if you're struggling to add a test case and I'll help!

### CODE REVIEW

**Comment by barnybug** [comment]
Thanks for your contribution.
We'd like to merge this pull request, but first please could you:

- ensure the existing tests pass (run: make test)

As it is, this only implements import, and cli53 actually errors on attempting to export. See failing test:

  Scenario: I can import a basic zone                                # internal/features/import.feature:7
    Given I have a domain "$domain"                                  # internal/features/import.feature:8
    When I run "cli53 import --file tests/basic.txt $domain"         # internal/features/import.feature:9

Error: exit status 1 Output: Error: InvalidChangeBatch: [Invalid Resource Record: FATAL problem: IncompleteOctalEscapeSequence (Value contains incomplete octal escape sequence) encountered with '"" "" "/urn:cid:.+@([^\.]+\.)(.*)$/\2/i"']
	status code: 400, request id: 2f0f557b-1be7-11e9-bad9-014a32bfe690

    Then the domain "$domain" export matches file "tests/basic.txt"  # internal/features/import.feature:10

**Comment by wfaulk** [comment]
It works for me?

```
% GOCACHE=off go test github.com/barnybug/cli53
ok  	github.com/barnybug/cli53	0.016s
```

**Comment by wfaulk** [comment]
First, despite being examples pulled directly from the NAPTR RFC, Route 53 didn't like the records I had put in the zone file.  I updated them to some that it did like.

I also added support for exporting NAPTR records, and it passes the test now.  However, Route 53 seems to return NAPTR records in an odd/incorrect format. It always quotes the last field (replacement), which seems to be wrong, unless the last field is just `.`, in which case it doesn't quote it.  The way I'm parsing it now works, but it's vaguely possible that Amazon might change something in the future that could break the regex.

(Also, go's regex support is pretty lousy. It doesn't support backreferences, which would have made this cleaner, and it claims to support non-capturing groups, but they get captured anyway.)

File: bind.go [hunk 1]
```diff
@@ -294,6 +306,31 @@ func ConvertRRSetToBind(rrset *route53.ResourceRecordSet) []dns.RR {
 				}
 				ret = append(ret, dnsrr)
 			}
+		case "NAPTR":
+			for _, rr := range rrset.ResourceRecords {
+				// parse value
+				naptrRE, err := regexp.Compile("^([[:digit:]]+) ([[:digit:]]+) \"([^\"]*)\" \"([^\"]*)\" \"([^\"]*)\" \"?([^\"]+)\"?$")
```
[the remark this answers is not in the document]
Comment (barnybug) [reply bind.go]: Also, this regexp can be written more legibly using raw strings (backticks), to cut down on the escaping required.

[the remark this answers is not in the document]
Comment (wfaulk) [reply bind.go]: Got it. Fixed.

File: bind.go [hunk 2]
```diff
@@ -294,6 +306,31 @@ func ConvertRRSetToBind(rrset *route53.ResourceRecordSet) []dns.RR {
...
+				naptr := naptrRE.FindStringSubmatch(*rr.Value)
+				fmt.Printf("naptr slice: %q\n", naptr)
```
[the remark this answers is not in the document]
Comment (wfaulk) [reply bind.go]: Whoops. Fixed.

**Comment by wfaulk** [comment]
Fixed your issues above and added a unit test NAPTR record

**Pull request merged**